samples/excel/89-preview-apis/range-hyperlink.yaml

```diff
@@ -10,121 +10,267 @@ script:
+                    cellRange.clear(Excel.ClearApplyTo.hyperlinks);
+
+                    // Update text format.
+                    cellRange.format.font.underline = Excel.RangeUnderlineStyle.none;
+                    cellRange.format.font.color = "#000000";
+                }
 
                 await context.sync();
+
+                OfficeHelpers.UI.notify("Cleared all hyperlinks and reset text format.");
             });
         }
 
         async function setup() {
             await Excel.run(async (context) => {
-                const sheet = await OfficeHelpers.ExcelUtilities
-                    .forceCreateSheet(context.workbook, "Sample");
 
-                const productsData1 = [
-                    ["Vegetables", "Qty", "Unit Price", "Total Price"],
-                    ["Potatoes", 10, 1.00, "=C3 * D3"],
-                    ["Peppers", 7, 2.50, "=C4 * D4"],
-                    ["Lettuce", 5, 1.50, "=C5 * D5"]
-                ];
+                const ordersSheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Orders");
+
+                const appleSheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Apple");
+
+                const bananaSheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Banana");
 
-                const range1 = sheet.getRange("B2:E5");
-                range1.values = productsData1;
-                range1.format.autofitColumns();
-                range1.format.fill.color = "lightgray";
+                const melonSheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Melon");
 
-                const header1 = range1.getResizedRange(-3, 0);
-                header1.format.fill.color = "yellow";
+                createOrdersData(ordersSheet, context);
 
-                const productsData2 = [
-                    ["Fruit", "Qty", "Unit Price", "Total Price"],
-                    ["Apples", 10, 2.00, "=C9 * D9"],
-                    ["Bananas", 5, 0.75, "=C10 * D10"],
-                    ["Melons", 8, 3.50, "=C11 * D11"]
-                ];
+                createAppleData(appleSheet, context);
 
-                const range2 = sheet.getRange("B8:E11");
-                range2.values = productsData2;
-                range2.format.autofitColumns();
-                range2.format.fill.color = "lightgray";
+                createBananaData(bananaSheet, context);
 
-                const header2 = range2.getResizedRange(-3, 0);
-                header2.format.fill.color = "lightblue";
+                createMelonData(melonSheet, context);
 
-                sheet.activate();
+                ordersSheet.activate();
 
                 await context.sync();
             });
         }
 
+        async function createOrdersData(ordersSheet: Excel.Worksheet, context: Office.Context) {
```

Comment:
All of the create*Data() methods have the `async `keyword, but they don't contain anything awaitable. Apparently, the compiler let's you get away with this, but I think it might confuse developers. Unless there's some reason I'm not aware of for having the `async `keyword on these methods, can we remove it?

Reply:
Good catch!  Originally, I was doing sheet creation in each of these functions (so I needed to pass-in a **context** object and include the word **async** in the signature)....but since I moved the sheet creation statements into the setup() function instead, the *Data() functions no longer need to be **async** or to accept a **context** object as an input parameter. I'll make these changes...thanks!